modules/imgproc/src/intelligent_scissors.cpp

```diff
@@ -625,8 +654,8 @@ struct IntelligentScissorsMB::Impl
                 CV_DbgCheckLE(cost_q, cost_r, "INTERNAL ERROR: sorted queue is corrupted");
 #endif
 
-                float cost = cost_q + local_cost(q, r);  // TODO(opt): compute partially until cost < cost_r
-                if (cost < cost_r)
+                float cost = cost_q;
+                if (add_local_cost(q, r, cost, cost_r))
```

Comment:
What is observed performance result? on which configuration?

Reply:
Updated the PR description with snippets from the `opencv_perf_imgproc` test run. Not sure w.r.t. configuration(s), this was on Mac with `cmake -DCMAKE_BUILD_TYPE=Release` build.